### PR TITLE
Fixes for native-image generation

### PR DESCRIPTION
**Pull request by aslesarenko**

### CODE REVIEW

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala [hunk 1]
```diff
@@ -24,4 +28,40 @@ object EncryptedSecret {
       Base16.encode(authTag), cipherParams)
   }
 
+  implicit object EncryptedSecretEncoder extends Encoder[EncryptedSecret] {
```
Comment (oskin1) [inline ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala]: Why adding this boilerplate -- encoder/decoder for this class is easily derived by circe?

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala [hunk 2]
```diff
@@ -24,4 +28,40 @@ object EncryptedSecret {
...
+
+  }
+
+//  def main(args: Array[String]): Unit = {
+//    val res = decode[EncryptedSecret](
+//      """{ "cipherText": "", "salt": "", "iv": "", "authTag": "", "cipherParams": { "prf":"abc", "c": 10, "dkLen": 100 }}""")
+//    val jsonRaw = res.right.get.asJson.noSpaces
+//    println(res.toString)
+//    println(jsonRaw)
+//  }
```
Comment (oskin1) [inline ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala]: Do you want to leave it in the code?

Comment (kushti) [reply ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala]: commented out code should be avoided I guess

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala (hunk 1 above)
Comment (aslesarenko) [reply ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala]: Unfortunately, that macros generates code not supported by native-image of GraalVM. 
This is the point of this PR, to make the code native-image compatible.

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala (hunk 2 above)
Comment (aslesarenko) [reply ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala]: removed

**Pull request merged**